### ISSUE TITLE
[bug]: Asterisk Log Files Not Rotating with Date

### ISSUE THREAD

**Issue by ramarajan222** [FreePBX/issue-tracker#220]
### FreePBX Version

FreePBX 17

### Issue Description

Hi Team,

The Asterisk log files are not rotating with the date. Currently, they create new files with names like full.1, full.2, whereas in the V16 system, the files are named with dates.
The "dateext" option is disabled by default in logger.conf on Debian systems, so we need to enable it to create log files with dates.

`
/var/log/asterisk# ls -l | grep full
-rw-r----- 1 asterisk asterisk         0 Jun  6 07:32 full
-rw-r----- 1 asterisk asterisk    260426 Jun  6 07:32 full.1
-rw-rw-r-- 1 asterisk asterisk   1751849 Jun  5 16:35 full.2

`

### Operating Environment

FreePBX V17

### Relevant log output

```shell
cat /etc/logrotate.conf
# see "man logrotate" for details

# global options do not affect preceding include directives

# rotate log files weekly
weekly

# keep 4 weeks worth of backlogs
rotate 4

# create new (empty) log files after rotating old ones
create

# use date as a suffix of the rotated file
#dateext

# uncomment this if you want your log files compressed
#compress

# packages drop log rotation information into this directory
include /etc/logrotate.d

# system-specific logs may also be configured here.
```

**Comment by dolesec** [comment]
Hi @ramarajan222 - in addition to the full log can you take a look at the backup-* logs as well as the queue_log*'s 

looking over my /var/log/asterisk those two stand out... 

```
-rw-rw-r-- 1 asterisk asterisk     49966 May 27 01:00 backup-09e5819f-1430-42ba-bd8a-ce0bca9378ce.log
-rw-r--r-- 1 asterisk asterisk     50636 Jun  6 01:00 backup-172e17d2-680c-4cb7-8d32-d2b5b5d4eafd.log
-rw-rw-r-- 1 asterisk asterisk     49966 May 26 01:00 backup-1da2b86b-9574-4a2d-923c-bfffe4cfe7c0.log
-rw-rw-r-- 1 asterisk asterisk     49543 May 19 01:00 backup-2593e9e5-77b0-4eca-9df2-bb43dcbbbf9d.log
-rw-rw-r-- 1 asterisk asterisk     49567 May 14 01:00 backup-29111399-4850-468c-a6f0-10414a71dee8.log
-rw-rw-r-- 1 asterisk asterisk     50178 May 13 06:50 backup-32ff79ed-b165-4407-af87-cdfab0a1d5bb.log
-rw-rw-r-- 1 asterisk asterisk     50182 May 13 06:48 backup-401ef249-fca3-449b-bf7d-2816bca23039.log
-rw-rw-r-- 1 asterisk asterisk     50178 May 13 06:37 backup-444cb473-39d3-4cc0-92d3-326ae73d5ff6.log
-rw-r--r-- 1 asterisk asterisk     50636 Jun  5 01:00 backup-451409ca-9eff-484c-a56f-0c6d8e2664c6.log
-rw-rw-r-- 1 asterisk asterisk     50636 Jun  4 01:00 backup-4aaaa904-db22-4abd-bd19-976c40a150ab.log
-rw-rw-r-- 1 asterisk asterisk     49966 May 29 01:00 backup-4d3de924-9903-4172-9131-556525dac748.log
-rw-rw-r-- 1 asterisk asterisk     49565 May 13 07:03 backup-5310d8c3-bf13-4348-85d3-9835f6d0e162.log
-rw-rw-r-- 1 asterisk asterisk     49541 May 22 09:22 backup-583f906d-164a-4097-b07b-aaa7ea5b4048.log
-rw-rw-r-- 1 asterisk asterisk     50633 Jun  3 01:00 backup-5b8b157c-6960-4334-87ee-f1194acb030e.log
-rw-rw-r-- 1 asterisk asterisk     50182 May 13 06:34 backup-60dc39e0-50f5-4da6-8a6c-9c8486b2a0bd.log
-rw-rw-r-- 1 asterisk asterisk     49545 May 20 01:00 backup-6939641f-7d19-475a-a377-10b30ec13210.log
-rw-rw-r-- 1 asterisk asterisk     49963 May 25 01:00 backup-756d8a13-35fa-4297-a024-f68e7c977c2f.log
-rw-rw-r-- 1 asterisk asterisk     49543 May 22 01:00 backup-798a30ae-2461-4dcd-a5d6-76f7c876e9ab.log
-rw-rw-r-- 1 asterisk asterisk     49565 May 17 07:55 backup-86595e73-a3e3-41ac-9139-8fb30ef2ebbf.log
-rw-rw-r-- 1 asterisk asterisk     49637 May 31 01:00 backup-88d9fd09-1b6f-49b0-afb0-fd2383cb5514.log
-rw-rw-r-- 1 asterisk asterisk     49567 May 17 01:03 backup-8fcf7a78-1f5b-4e03-a47e-e8849e06292c.log
-rw-rw-r-- 1 asterisk asterisk     49567 May 16 01:03 backup-97dd60d7-cbc5-46c2-8b76-55816440b8a7.log
-rw-rw-r-- 1 asterisk asterisk     49567 May 17 07:25 backup-aa47986d-6b51-40e4-94f0-df490d701bb0.log
-rw-rw-r-- 1 asterisk asterisk     49963 May 28 01:00 backup-aab30142-f8e4-46bb-935e-fb18b45aa26d.log
-rw-rw-r-- 1 asterisk asterisk     49963 May 23 01:00 backup-b21d7690-18d2-48cd-b137-a7ca83d1ea1a.log
-rw-rw-r-- 1 asterisk asterisk     49963 May 24 01:00 backup-b86bf345-8851-482a-97c7-0c4dcb63f689.log
-rw-rw-r-- 1 asterisk asterisk     49545 May 21 01:00 backup-bf7e6814-4f90-40c8-b2bf-1200cf99fcee.log
-rw-rw-r-- 1 asterisk asterisk     50178 May 13 06:52 backup-c82dbd4d-a8a1-4f76-af24-efd5389fb1e1.log
-rw-rw-r-- 1 asterisk asterisk     49635 Jun  2 01:00 backup-d361a62f-c7fd-4260-a299-de1c8c0cd140.log
-rw-rw-r-- 1 asterisk asterisk     49565 May 17 07:53 backup-e2c998dc-d46d-4812-aeeb-9b96647b6e94.log
-rw-rw-r-- 1 asterisk asterisk     49567 May 13 07:20 backup-e6b723d5-4ac5-4a6f-8e73-efe20bda43ec.log
-rw-rw-r-- 1 asterisk asterisk     49567 May 17 07:56 backup-e9f37319-5dfb-4fe1-95d8-af5998ae91f7.log
-rw-rw-r-- 1 asterisk asterisk     49635 Jun  1 01:00 backup-ed0e5204-708e-4d62-978e-b2a11c6bbfcd.log
-rw-rw-r-- 1 asterisk asterisk     49966 May 30 01:00 backup-f5f29b28-e9f1-40bf-bbbf-fc340266c7bd.log
-rw-rw-r-- 1 asterisk asterisk     49567 May 13 07:04 backup-f9e39141-e287-4739-b486-01003ed47afb.log
-rw-rw-r-- 1 asterisk asterisk    117988 Jun  2 01:00 backup.log
```


```
-rw-r----- 1 asterisk asterisk        80 Jun  6 00:00 queue_log
-rw-r----- 1 asterisk asterisk        80 Jun  5 00:00 queue_log.1
-rw-rw-r-- 1 asterisk asterisk        80 May 27 00:00 queue_log.10
-rw-rw-r-- 1 asterisk asterisk        80 May 26 00:00 queue_log.11
-rw-rw-r-- 1 asterisk asterisk       120 May 25 07:59 queue_log.12
-rw-rw-r-- 1 asterisk asterisk        80 May 24 00:00 queue_log.13
-rw-rw-r-- 1 asterisk asterisk        80 May 23 00:00 queue_log.14
-rw-rw-r-- 1 asterisk asterisk       158 May 22 09:15 queue_log.15
-rw-rw-r-- 1 asterisk asterisk       158 May 21 07:12 queue_log.16
-rw-rw-r-- 1 asterisk asterisk        80 May 20 00:00 queue_log.17
-rw-rw-r-- 1 asterisk asterisk        80 May 19 00:00 queue_log.18
-rw-rw-r-- 1 asterisk asterisk        38 May 18 13:37 queue_log.19
-rw-rw-r-- 1 asterisk asterisk       718 Jun  4 13:18 queue_log.2
-rw-rw-r-- 1 asterisk asterisk       514 May 17 17:21 queue_log.20
-rw-rw-r-- 1 asterisk asterisk       236 May 16 21:36 queue_log.21
-rw-rw-r-- 1 asterisk asterisk       116 May 15 19:55 queue_log.22
-rw-rw-r-- 1 asterisk asterisk       438 May 14 20:18 queue_log.23
-rw-rw-r-- 1 asterisk asterisk      1542 May 13 21:30 queue_log.24
-rw-rw-r-- 1 asterisk asterisk        80 Jun  3 00:00 queue_log.3
-rw-rw-r-- 1 asterisk asterisk       160 Jun  2 21:33 queue_log.4
-rw-rw-r-- 1 asterisk asterisk       240 Jun  1 20:43 queue_log.5
-rw-rw-r-- 1 asterisk asterisk        80 May 31 00:00 queue_log.6
-rw-rw-r-- 1 asterisk asterisk       158 May 30 06:51 queue_log.7
-rw-rw-r-- 1 asterisk asterisk        80 May 29 00:00 queue_log.8
-rw-rw-r-- 1 asterisk asterisk        80 May 28 00:00 queue_log.9
```

**Comment by ramarajan222** [comment]
Hi @dolesec ,

The Queue_log log is already added to logrotate, and it will keep a maximum of 30 files. We need to address the backup log issue and will create a separate Jira ticket for that.

`
cat /etc/logrotate.d/asterisk
/var/log/asterisk/queue_log {
        daily
        missingok
        rotate 30
        notifempty
        sharedscripts
        create 0640 asterisk asterisk
        su asterisk asterisk
        postrotate
                /usr/sbin/asterisk -rx 'logger reload' > /dev/null 2> /dev/null
        endscript
}


`


You can run the command below on your system to create the log files with dates. Additionally, we have included this command in the installation script to ensure it is applied to new systems.

`
sed -i 's/^#dateext/dateext/' /etc/logrotate.conf

`

**Comment by lgaetz** [comment]
Hi @ramarajan222 

When you see asterisk log files being rotated like full, full.1, full.2 etc. that is Asterisk doing the rotation not logrotate. But asterisk logfile rotation should be disabled by default on a distro install:
![image](https://github.com/FreePBX/issue-tracker/assets/868756/0d962eef-05f6-4f04-9754-28de3c336818)

On one of my 17 test systems, I'm missing the logger.conf symlink
```
# ll /etc/logger.conf
ls: cannot access '/etc/logger.conf': No such file or directory
```
But on another 17 system, I do have the symlink. As long as the symlink is misisng, none of the logging GUI settings will be applied. 

edit: **This is a new bug in logfiles module** ver. 17.0.3.2, when upgrading to this version, the logger.conf symlink gets removed.

**Comment by ramarajan222** [comment]
Hi @lgaetz 

Asterisk log rotation setting is set to "None " on the latest asterisk logfiles module (17.0.3.2)

**Comment by lgaetz** [comment]
Hi Ram, see my edit above, ver. 17.0.3.2 has a bug, /etc/astersisk/logger.conf symlink is missing

**Comment by dolesec** [comment]
seeing same

**Comment by ramarajan222** [comment]
Hi @lgaetz @dolesec We reverted the 17.0.3.2 fix and the logfiles V17.0.3.3 should fix the symlink issue. thanks
